include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -178,35 +179,19 @@ struct __vector_iter_distinguishes_by_allocator<
 };
 
 template <typename Iter, typename Void = void>
-struct __is_known_usm_vector_iter_impl : std::false_type
+struct __is_known_usm_vector_iter : std::false_type
 {
 };
 
 template <typename Iter>
-struct __is_known_usm_vector_iter_impl<
+struct __is_known_usm_vector_iter<
     Iter, std::enable_if_t<std::conjunction<
               std::disjunction<std::is_same<Iter, oneapi::dpl::__internal::__usm_shared_alloc_vec_iter<Iter>>,
                                std::is_same<Iter, oneapi::dpl::__internal::__usm_host_alloc_vec_iter<Iter>>>,
               oneapi::dpl::__internal::__vector_iter_distinguishes_by_allocator<Iter>>::value>> : std::true_type
 {
 };
```

Comment:
I wonder if the same result would be get by simply declaring `__is_known_usm_vector_iter` as an alias to the logical expression in the specialization, starting from `std::conjunction`. Also maybe the same technic can be used with `__vector_iter_distinguishes_by_allocator`? And the last, do we need those as types, or constexpr bool variables would be enough?

Reply:
Good point.  I had this originally but ran into some errors with permutation iterator index maps.  However, this is a problem with the way we define `permutation_iterators` with relation to `is_passed_directly`, rather than anything else.  

I've resolved that problem, and now simplified these to `constexpr bool`.